### PR TITLE
Add flag to select enabled CPU targets

### PR DESCRIPTION
**Pull request by RoboTux**
Default flag to current set of support targets to keep the default of enabling all CPU targets. This fixes issue [#59595](https://github.com/tensorflow/tensorflow/issues/59595) as well as some confusiong around Host Vs Target.

### CODE REVIEW

**Comment by RoboTux** [comment]
Note on testing: I've run ./tensorflow/tools/ci_build/ci_build.sh CPU.ARM64 bazel test on AArch64 on //tensorflow/core/... //tensorflow/tsl/... and //tensorflow/compiler/... and there was no regression.

**Comment by ddunl** [comment]
This looks good to me, but I'm not sure I'm the right person to review. I'll add @MichaelHudgins and @akuegel, hopefully one of them can review (or will know who to assign). Thanks for the PR!

File: tensorflow/tsl/BUILD [hunk 1]
```diff
@@ -1,4 +1,5 @@
-load("tsl.bzl", "if_google", "if_oss")
+load("tsl.bzl", "if_google", "if_oss", "enable_targets_flag")
+load("//third_party/llvm:setup.bzl", "supported_targets")
```
Comment (akuegel) [inline tensorflow/tsl/BUILD:2]: This will probably require some internal adjustments, so we would need to find someone to work on this. Internally we don't use setup.bzl.

**Review by akuegel — COMMENTED** [review]
In general I think this change is good, but it might be blocked by not working internally. This is something that probably can be fixed, but it needs someone to work on this.

File: tensorflow/tsl/BUILD (hunk 1 above)
Comment (RoboTux) [reply tensorflow/tsl/BUILD:2]: I think initially I tried to do the reverse and have third_party/llvm:setup.bzl use supported_targets defined in the core (or tsl, I can't quite remember) but there was some dependency issue. I'd need to try again.

File: third_party/llvm/setup.bzl [hunk 2]
```diff
@@ -3,7 +3,7 @@
 load("@llvm-raw//utils/bazel:configure.bzl", "llvm_configure", "llvm_disable_optional_support_deps")
 
 # The subset of LLVM targets that TensorFlow cares about.
-_LLVM_TARGETS = [
+supported_targets = [
```
Comment (akuegel) [inline third_party/llvm/setup.bzl:6]: It would significantly simply internal adjustments if you could rename this to "llvm_targets".

**Comment by gbaned** [comment]
Hi @RoboTux Can you please check @akuegel's comments and keep us posted ? Thank you!

**Comment by RoboTux** [comment]
> Hi @RoboTux Can you please check @akuegel's comments and keep us posted ? Thank you!

TL;DR: saw that, I was trying to understand the current default behavior, will rework the patch accordingly and change it to default to enable only the host architecture.

Yes sorry I saw that but I realized in https://github.com/tensorflow/tensorflow/issues/59595#issuecomment-1473725625 that I erroneously thought all targets are enabled by default currently which is not the case, only X86 and the host CPU are enabled by default. I delayed a bit to find the time to analyze why despite llvm_targets having all targets in @llvm-project it did not cause all targets to be enabled. It's down to the config_settings operating on the --cpu value in //tensorflow/tsl which I thought did not work as intended because of how it's documented in bazel as having an empty default:

https://bazel.build/reference/command-line-reference#flag--cpu

Best regards,
Thomas

**Review by akuegel — APPROVED** [review]

**Comment by akuegel** [comment]
Ah, it seems there are some merge conflicts that need to be resolved.

**Comment by RoboTux** [comment]
What commit message will be used when this land after I solve the conflicts? Will it use the original message or what I last forced pushed?

**Comment by gbaned** [comment]
Hi @RoboTux Can you please resolve conflicts? Thank you!
